crates/wasmtime/src/store.rs

```diff
@@ -1856,7 +1856,14 @@ unsafe impl<T> wasmtime_runtime::Store for StoreInner<T> {
 
     fn new_epoch(&mut self) -> Result<u64, anyhow::Error> {
         return match &self.epoch_deadline_behavior {
-            &EpochDeadline::Trap => Err(anyhow::Error::new(EpochDeadlineError)),
+            &EpochDeadline::Trap => {
+                let trap = Trap::new_wasm(
```

Comment:
It seems that this call could be simplified in this case by exposing a `Trap::from_trap_code` function in the Trap implementation, which would take in the `TrapCode` and apply the corresponding defaults for the `pc` and `backtrace`.

Didn't introduce this change right away since this is the only use case but I can totally do it if you think it's worth it.

Reply:
This seems reasonable enough to me for now, if this pops up more often we can make a helper for it